packages/canary-web-components/src/utils/date-helpers.ts

```diff
@@ -0,0 +1,90 @@
+import { IcDateFormat } from "../utils/types";
+
+export const isMonthFirstFormat = (dateString: string) => {
+  const monthFirstFormat =
+    /^(0[1-9]|1[0-2])(\/|-)(0[1-9]|[12][0-9]|3[01])(\/|-)\d{4}$/;
+  return monthFirstFormat.test(dateString);
+};
+
+export const isDayFirstFormat = (dateString: string) => {
+  const dayFirstFormat =
+    /^(0[1-9]|[12][0-9]|3[01])(\/|-)(0[1-9]|1[0-2])(\/|-)\d{4}$/;
+  return dayFirstFormat.test(dateString);
+};
+
+export const createDate = (date: string | Date, dateFormat: IcDateFormat) => {
+  let newDate;
+  if (isDateOrEpoch(date)) {
+    if (typeof date === "string") {
```

Comment:
dunno how you feel about ternary operators but this if/else could be
```js
newDate = (typeof date === "string") ? new Date(date) : date;
```
i think.